### PR TITLE
#163223218 User registration

### PR DESCRIPTION
**Pull request by Philipotieno**
#### What does this PR do?

- Enables user to register by providing the first name, last name, email, phone and password
- Pass failing tests

#### Description of Task to be completed?

- Add an endpoint to request for user data
- Make sure user data is validated
- One can one use a username once

#### How should this be manually tested?

- Git clone this repo [https://github.com/Philipotieno/Questioner-API](https://github.com/Philipotieno/Questioner-API)
`$ https://github.com/Philipotieno/Questioner-API.git`
- cd in directory
`$ cd Questioner-API `

- Install virtualenv:

`$ pip install virtualenv`
- Create a virtual environment in the root directory:

$ virtualenv -name of virtualenv-

Note: If you do not have python3 installed globally, please run this command when creating a virtual environment:

`$ virtualenv -p python3 -name of virtualenv-`

- Activate the virtualenv:

`$ source name of virtualenv/bin/activate`

- Install the requirements of the project:

`$ pip install -r requirements.txt`

- Create a file in the root directory called .env and add the lines below

`export FLASK_APP="run.py"`
`export FLASK_ENV=development`
`export SECRET_KEY="some random string"`
`export APP_SETTINGS="development"`

`export DB_URL="postgresql://localhost/questioner"`
`export DB_HOST="localhost"`
`export DB_NAME="database_name"`
`export TEST_DB="testdb"`
`export DB_USERNAME="user role"`
`export DB_PASSWORD="password"`

- Activate the env variables:

`$ source .env`

- Run the application:

`$ python manage.py`
`$ python run.py`
-  Test this the endpoint using postman
 `POST /api/v1/user/register`
- To run tests:
`$ pytest`

####PT STORY
[https://www.pivotaltracker.com/n/projects/2236065/stories/163046495](https://www.pivotaltracker.com/n/projects/2236065/stories/163046495)

####Screenshots
- Tests
tests

- Register user
vin

- Register user with an existing username
reg user

### CODE REVIEW

**Comment by bencyn** [comment]
Fix typo `provising` in the first section

**Comment by Philipotieno** [comment]
Thanks, @bencyn Updated it

File: app/tests/v2/base_v2.py [hunk 1]
```diff
@@ -83,30 +82,25 @@ def setUp(self):
...
+        # self.login = self.client.post(
+        #     '/api/v2/auth/login',
+        #     data=json.dumps(self.user_1),
+        #     content_type='application/json')
 
-        self.create_meetup = self.client.post(
-            '/api/v1/meetups',
-            data=json.dumps(self.meetup_1),
-            content_type='application/json')
+        # self.create_meetup = self.client.post(
```
Comment (migot01) [inline app/tests/v2/base_v2.py:90]: remove code that is commented out before pushing it

Comment (Philipotieno) [reply app/tests/v2/base_v2.py:90]: Thanks will correct that in my next PR

**Pull request merged**